lib/echo/echo_client.rb

```diff
@@ -34,10 +34,39 @@ def get_phones(user_id, token)
       get("/echo-rest/users/#{user_id}/phones.json", {}, token_header(token))
     end
 
-    def get_preferences(user_id, token)
+    def general_contact_valid?(contact)
+      return false if contact.blank? || contact['phones'].blank? || contact['address'].blank?
+      true
+    end
+
+    def get_preferences(user_id, token, client, access_token)
       response = get("/echo-rest/users/#{user_id}/preferences.json", {}, token_header(token))
       if response.status == 404
         response = update_preferences(user_id, {preferences: {}}, token)
+      elsif response.status == 200 && !general_contact_valid?(response.body['preferences']['general_contact'])
```

Comment:
Not quite sure what you mean...
Firstly, there are two routes that can hit this ```def get_preferences``` method:
 - From an AJAX (```account.coffee```) when user is on the contact_info page
   - It will go through ```users_controller#get_preferences``` which contacts URS and gets the user profile there.
 - When user tries to place an order
   - if it returns a 404, the ```general_contact``` will be constructed in ```def create_order```
   - if it returns a 200 but without a valid ```general_contact```, the contact will be constructed in this method and another PUT will be sent out.